### DISCUSSION TITLE
issue: update nested object value entire

### DISCUSSION THREAD

**Discussion by yjkwon07** [react-hook-form/react-hook-form#9164] [Q&A] [answered]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-object-change-lcl7b2?file=/src/index.js

### Steps to reproduce

When I use setValue for object reset, not trigger register value. 😕

```jsx
const defaultValues = {
  name: { site: "vanilla", code: "ss" }
};

...

<button
  className="button buttonBlack"
  type="button"
  onClick={() => {
    setValue("name", undefined);
  }}
>

...

<input {...register("name.site")} />
```

### Expected behaviour

I found solution1 just update key name value. It's fine.

```jsx

<button
  className="button buttonBlack"
  type="button"
  onClick={() => {
    setValue("name.site", undefined);
  }}
>

<input {...register("name.site")} />
```

- but, I want entire object reset, trigger too.. 😢

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Please refer to the doc: 

https://react-hook-form.com/api/useform/setvalue

> The value for the field. This argument is required and can not be undefined.

you can consider to use reset if you want to bulk update form values.

**Comment by yjkwon07** [comment, marked as answer]
ok, thanks!

So, this code is incorrect?

```jsx
setValue("name.site", undefined);
```

---

(Same as above situation) When I use onChange in useController object, not trigger.

[code](https://codesandbox.io/s/react-hook-form-object-change-forked-63e391?file=/src/index.js)

and also use setValue instead of onChange in useController, not woking.
- If useController code line remove, that's fine.

```jsx
 const { control, register, handleSubmit, setValue } = useFormContext();

  // const { field } = useController({ control, name: "name" }); // if remove line setValue working
  renderCount++;

  const handleResetField = useCallback(() => {
    setValue("name", { site: "" });
  }, [setValue]);
```

when just bulk update, I have to use reset?

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-object-change-controller-forked-38l2yt?file=/src/index.js

```tsx
const handleResetField = () => {
  setValue("name.site", "");
  setValue("name.code", "");

  setValue('name', {
    site: '',
    code: ''
  })
};
```

**Reply by yjkwon07** [reply]
> (Same as above situation) When I use onChange in useController object, not trigger.

```jsx
const { control, register, handleSubmit, setValue } = useFormContext();

const { field } = useController({ control, name: "name" });

const handleResetField = useCallback(() => {
   field.onChange({ site: "", code: "" });
}, [field, setValue]);

...

 <Controller
    control={control}
    name="name.site"
    render={({ field: { value, onChange } }) => (
      <input value={value} onChange={onChange} />
    )}
  />
  <input {...register("name.code")} />
```

![image](https://user-images.githubusercontent.com/31876632/194518803-2ea73484-e249-4ab0-90e1-0508d73732de.png)

- Controller "name.site" => not trigger.
- register "name.code" => trigger.
- why? Controller not trigger.

**Reply by bluebill1049** [reply]
> why? Controller not trigger.

`const { field } = useController({ control, name: "name" });`

because `useController` only targets that `useController` instance, it has no context of other register input. a `useController` at the root doesn't mean it controls the entire form. you should consider `useController` as its own identity for the input. It has its own state scope.

**Reply by yjkwon07** [reply]
thanks for the quick reply! 😄

I understand useController, setValue, this time!!

**Reply by bluebill1049** [reply]
I am very glad!
